docs/developer/style.md

```diff
@@ -0,0 +1,113 @@
+# RDycore (C) Style Guide
+
+Here we provide a simple style guide for C code within the RDycore library. This
+style guide applies mainly to source code within the RDycore C library itself.
+The corresponding Fortran library is a thin hand-written wrapper around the C
+library that doesn't leave much room for "style," so there is no Fortran style
+guide.
+
+This style guide is very short compared to many others, since C is a very small
+language and most formatting conventions (whitespace, lengths of lines,
+placement of curly braces) are enforced via tools and not humans. Here, we focus
+mainly on those conventions that need the attention of RDycore team members and
+other contributors.
+
+## Style Conventions
+
+RDycore's conventions are based on those in the
+[PETSc Style and Usage Guide](https://petsc.org/release/developers/style/),
+with the following deviations:
+
+* we allow the use of variable-length arrays (VLAs)
+* we use standard header guards instead of the nonstandard `#pragma once`
+  mechanism
+* we prefer C++-style code comments (`//`) to C-style comments (`/* */`) even
+  for multi-line comments
+* we use `if (foo)` instead of `if (foo != NULL)` and `if (cond)` instead of
```

Comment:
First `if (foo)` should be `if (!foo)`.

Reply:
Ah, but it shouldn't, because `foo != NULL` is a `true` value, and `if (foo)` tests the "truthiness" of `foo`. I'll add more examples for clarity.